spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/export/prometheus/PrometheusProperties.java

```diff
@@ -116,6 +118,21 @@ public Pushgateway getPushgateway() {
 		 */
 		private Map<String, String> groupingKey = new HashMap<>();
 
+		/**
+		 * Enable publishing via a Prometheus Pushgateway with Basic Auth.
+		 */
+		private Boolean authEnabled = false;
```

Comment:
We don't have such `enabled` flag. What we usually do is to check that the username is set, something like `StringUtils.hasText` can help you with that.

Reply:
you mean check this value in `PrometheusMetricsExportAutoConfigurationTests`?

Reply:
No. I mean that we don't enable authentication based on a flag but based on the fact that the `username` property is set.